storage: TestStoreRangeMergeRHSLeaseExpiration failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/15e96b9fc4065e936eaf4f3f744cd75b41209261

Parameters:
```
TAGS=
GOFLAGS=-race
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=791000&tab=buildLog

```
=== RUN   TestStoreRangeMergeRHSLeaseExpiration
I180722 08:09:02.633015 50631 gossip/gossip.go:363  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:34369" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > 
W180722 08:09:02.726958 50631 gossip/gossip.go:1325  [n2] no incoming or outgoing connections
I180722 08:09:02.729294 50746 gossip/client.go:129  [n2] started gossip client to 127.0.0.1:34369
I180722 08:09:02.785318 50631 storage/store.go:1480  [s2] [n2,s2]: failed initial metrics computation: [n2,s2]: system config not yet available
I180722 08:09:02.788714 50631 gossip/gossip.go:363  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:46749" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > 
I180722 08:09:02.815139 50631 storage/replica_command.go:279  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "b" [r2]
I180722 08:09:02.884092 50631 storage/replica_raftstorage.go:520  [s1,r2/1:{b-/Max}] generated preemptive snapshot fb1da656 at index 11
I180722 08:09:02.884589 50631 rpc/nodedialer/nodedialer.go:90  [s1,r2/1:{b-/Max}] connection to n2 established
I180722 08:09:02.902310 50631 storage/store_snapshot.go:605  [s1,r2/1:{b-/Max}] streamed snapshot to (n2,s2):?: kv pairs: 40, log entries: 1, rate-limit: 2.0 MiB/sec, 17ms
I180722 08:09:02.904531 50834 storage/replica_raftstorage.go:737  [s2,r2/?:{-}] applying preemptive snapshot at index 11 (id=fb1da656, encoded size=7277, 1 rocksdb batches, 1 log entries)
I180722 08:09:02.907151 50834 storage/replica_raftstorage.go:743  [s2,r2/?:{b-/Max}] applied preemptive snapshot in 2ms [clear=0ms batch=0ms entries=0ms commit=1ms]
I180722 08:09:02.910698 50631 storage/replica_command.go:743  [s1,r2/1:{b-/Max}] change replicas (ADD_REPLICA (n2,s2):2): read existing descriptor r2:{b-/Max} [(n1,s1):1, next=2]
I180722 08:09:02.924734 50631 storage/replica.go:3613  [txn=e82f1934,s1,r2/1:{b-/Max}] proposing ADD_REPLICA((n2,s2):2): updated=[(n1,s1):1 (n2,s2):2] next=3
I180722 08:09:02.938711 50776 rpc/nodedialer/nodedialer.go:90  connection to n1 established
I180722 08:09:02.962760 50631 storage/replica_range_lease.go:554  transferring lease to s2
I180722 08:09:02.975030 50772 storage/replica_proposal.go:222  [s2,r2/2:{b-/Max}] new range lease repl=(n2,s2):2 seq=2 start=0.000000123,182 epo=1 pro=0.000000123,183 following repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,4 pro=0.000000123,5
I180722 08:09:02.983179 50631 storage/replica_range_lease.go:617  done transferring lease to s2: <nil>
I180722 08:09:03.042259 50886 storage/replica_command.go:406  [s1,r1/1:{/Min-b}] initiating a merge of r2:{b-/Max} [(n1,s1):1, (n2,s2):2, next=3] into this range
I180722 08:09:03.094622 50631 storage/client_test.go:1228  test clock advanced to: 1.800000125,0
I180722 08:09:03.117878 50860 util/stop/stopper.go:473  quiescing; tasks left:
1      [async] kv.TxnCoordSender: heartbeat loop
I180722 08:09:03.118478 50861 util/stop/stopper.go:473  quiescing; tasks left:
1      [async] wait-for-merge
W180722 08:09:03.337284 50868 storage/raft_transport.go:575  while processing outgoing Raft queue to node 1: EOF:
--- FAIL: TestStoreRangeMergeRHSLeaseExpiration (0.84s)
	client_merge_test.go:851: heartbeat failed on epoch increment

```

--- a/pkg/storage/client_test.go
+++ b/pkg/storage/client_test.go
@@ -1207,7 +1207,13 @@ func (m *multiTestContext) heartbeatLiveness(ctx context.Context, store int) err
 	if err != nil {
 		return err
 	}
-	return nl.Heartbeat(ctx, l)
+
+	for r := retry.StartWithCtx(ctx, retry.Options{MaxRetries: 5}); r.Next(); {
+		if err = nl.Heartbeat(ctx, l); err != storage.ErrEpochIncremented {
+			break
+		}
+	}
+	return err
 }
 
 // advanceClock advances the mtc's manual clock such that all

